torch/nn/modules/transformer.py

```diff
@@ -214,8 +218,8 @@ def forward(self, src: Tensor, tgt: Tensor, src_mask: Optional[Tensor] = None, t
     @staticmethod
     def generate_square_subsequent_mask(
             sz: int,
-            device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
-            dtype: torch.dtype = torch.get_default_dtype(),
+            device: Optional[torch.device] = None,
```

Comment:
This always calls into `_generate_square_subsequent_mask` so we don't need to handle the `None`